wire-runtime/src/main/java/com/squareup/wire/TypeAdapter.java

```diff
@@ -28,26 +28,50 @@
 
   final WireType type;
 
-  protected TypeAdapter(WireType type) {
+  TypeAdapter(WireType type) {
     this.type = type;
   }
 
-  /** The serialized size of non-null {@code value}. */
-  public abstract int serializedSize(E value);
+  /**
+   * The size of the non-null data {@code value}. This does not include the size required for
+   * a length-delimited prefix (should the type require one).
```

Comment:
True, but the context never matters for type adapters. It always wants the prefix to be accounted for (or in this case, ignored). They remain blissfully unaware of the root message nonsense.